### PR TITLE
:seedling: add querying options to list and watch

### PR DESCRIPTION
**Pull request by skeeey**
## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/cloudevents/work/agent/client/manifestwork.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (c *ManifestWorkAgentClient) Get(ctx context.Context, name string, opts met
 
 func (c *ManifestWorkAgentClient) List(ctx context.Context, opts metav1.ListOptions) (*workv1.ManifestWorkList, error) {
 	klog.V(4).Infof("list manifestworks")
-	works, err := c.watcherStore.List(opts)
+	works, err := c.watcherStore.List(c.clusterName, opts)
```
Comment (qiujian16) [inline pkg/cloudevents/work/agent/client/manifestwork.go:73]: does it support all?

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go:73]: yes, it does

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go:73]: handle here https://github.com/open-cluster-management-io/sdk-go/pull/60/files#diff-74d6100c69d0c83ceae8a3b90070967958787cbc0106de6ec2cdd0157828bb40R97

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go:73]: but for agent part, the cluster name always should have a given value

File: pkg/cloudevents/work/agent/client/manifestwork.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ import (
 type ManifestWorkAgentClient struct {
 	cloudEventsClient *generic.CloudEventAgentClient[*workv1.ManifestWork]
 	watcherStore      store.WorkClientWatcherStore
+	namespace         string
```
Comment (qiujian16) [inline pkg/cloudevents/work/agent/client/manifestwork.go:29]: I think namespace is the same as clusterName? why do we need two var?

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go:29]: The clusterName is initialized when we create this client, when we build this client, we will start to subscribe to a given cluster

namespace is set when we use this client, e.g. `client.namespace("aaaa").list(...)`

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go:29]: it should be ok to remove `clusterName`,  if we give a wrong namespace for client, the related work will not be found

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go:29]: updated to only use namespace

Comment (qiujian16) [reply pkg/cloudevents/work/agent/client/manifestwork.go:29]: add a comment here since it should be the same as clusterName.

Comment (skeeey) [reply pkg/cloudevents/work/agent/client/manifestwork.go]: added

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**